### DISCUSSION TITLE
Want to receive user name and bio when using 'Search users' API

### DISCUSSION THREAD

**Discussion by seenewmax** [community/community#115229] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

https://docs.github.com/ko/rest/search/search?apiVersion=2022-11-28#search-users

https://api.github.com/search/users?q=Q
This user searching API returns like below.

{
  "total_count": 12,
  "incomplete_results": false,
  "items": [
    {
      "login": "mojombo",
      "id": 1,
      "node_id": "MDQ6VXNlcjE=",
      "avatar_url": "https://secure.gravatar.com/avatar/25c7c18223fb42a4c6ae1c8db6f50f9b?d=https://a248.e.akamai.net/assets.github.com%2Fimages%2Fgravatars%2Fgravatar-user-420.png",
      "gravatar_id": "",
...
    }
  ]
}

Only contains login(=mojombo) now.
Need to request again to get more info.
https://api.github.com/users/mojombo

{
  "login": "mojombo",
  "id": 1,
  "node_id": "MDQ6VXNlcjE=",
  "avatar_url": "https://avatars.githubusercontent.com/u/1?v=4",
  "gravatar_id": "",
  "url": "https://api.github.com/users/mojombo",
...
  "name": "Tom Preston-Werner",
  "company": "@chatterbugapp, @redwoodjs, @preston-werner-ventures ",
  "blog": "http://tom.preston-werner.com",
  "location": "San Francisco",
  "email": null,
  "hireable": null,
  "bio": null,
  "twitter_username": "mojombo",
  "public_repos": 66,
  "public_gists": 62,
  "followers": 23830,
  "following": 11,
  "created_at": "2007-10-20T05:24:19Z",
  "updated_at": "2024-03-18T21:50:22Z"
}

Now I can get name and bio.
When searching user on Github app, it seems it directly fetches name and bio info.
Wonder if I can't to do so?

**Comment by LangLangBart** [comment, marked as answer]
> When searching user on Github app, it seems it directly fetches name and bio info. Wonder if I can't to do so?

Yes, you can achieve this by using the `GraphQL API` rather than the `Rest API`.

The `GraphQL API`[^1][^2] allows you to make a single specific request and receive exactly what you ask for.


```graphql
query ($query: String = "user:ashtom") {
  search(query: $query, type: USER, first: 100) {
    nodes {
      ... on User {
        bio
        login
      }
    }
  }
}
```

```json
{
  "data": {
    "search": {
      "nodes": [
        {
          "bio": "CEO @github",
          "login": "ashtom"
        }
      ]
    }
  }
}
```

[^1]: [About the GraphQL API - GitHub Docs](https://docs.github.com/en/graphql/overview/about-the-graphql-api#why-github-is-using-graphql)
[^2]: [GitHub - ByteByteGoHq/system-design-101: Explain complex systems using visuals and simple terms.](https://github.com/ByteByteGoHq/system-design-101#rest-api-vs-graphql)
